atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -151,15 +153,17 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
 
     @Override
     public TransactionAndImmutableTsLock setupRunTaskWithConditionThrowOnConflict(PreCommitCondition condition) {
-        StartAtlasDbTransactionResponse transactionResponse = timelockService.startAtlasDbTransaction(
-                IdentifiedTimeLockRequest.create());
+        StartIdentifiedAtlasDbTransactionResponse transactionResponse
+                = timelockService.startIdentifiedAtlasDbTransaction(
+                        StartIdentifiedAtlasDbTransactionRequest.createForRequestor(clientId));
         try {
             LockToken immutableTsLock = transactionResponse.immutableTimestamp().getLock();
             long immutableTs = transactionResponse.immutableTimestamp().getImmutableTimestamp();
             recordImmutableTimestamp(immutableTs);
 
-            cleaner.punch(transactionResponse.freshTimestamp());
-            Supplier<Long> startTimestampSupplier = Suppliers.ofInstance(transactionResponse.freshTimestamp());
+            cleaner.punch(transactionResponse.startTimestampAndPartition().timestamp());
+            Supplier<Long> startTimestampSupplier = Suppliers.ofInstance(
+                    transactionResponse.startTimestampAndPartition().timestamp());
 
             SnapshotTransaction transaction = createTransaction(immutableTs, startTimestampSupplier,
```

Comment:
This and a couple of tests seem to be the only places the method is called, including large internal product. In each instance we have a supplier of instance, so why don't we change it to just a `long`? As a bonus, you can reorder the statements above to then `cleaner.punch(startTs)`

Reply:
Discussed offline - we can push this one layer in, but unfortunately in `SnapshotTransaction` you still need the supplier since read only transactions grab a timestamp lazily